### DISCUSSION TITLE
actix-web rejects message with 400, any way to log that?

### DISCUSSION THREAD

**Discussion by bburdette** [actix/actix-web#3054] [General] [answered]
Normally when the web server rejects a request, I can look at the request/reply in the browser debugger and get a clue about why it didn't work.  

However right now I'm trying to run a tauri android app - it sends a POST to my web server, and receives a 400 in reply.  I can't open the browser in this case.  When the message is received by the web server, it looks like it never gets as far as my regular handler function for that endpoint.  

Is there a way to log request/reply pairs so I can look at them, or to determine in some other way what the reason for the 400 was?

**Comment by moy2010** [comment, marked as answer]
You could either create a middleware for logging purposes or leverage the ecosystem and use something like tracing-actix-web.

**Reply by bburdette** [reply]
I switched to tracing-actix-web.  I didn't see how to log the contents of the request and reply messages, but I did at least get a better error message that was helpful in tracking down the issue.
